crates/test-programs/wasi-tests/src/bin/path_link.rs

```diff
@@ -1,7 +1,34 @@
+#![allow(warnings)]
 use more_asserts::assert_gt;
 use std::{env, process};
 use wasi_tests::{create_file, open_scratch_directory};
 
+// RAII adapter for path_open. On Windows we need to explicitly close the fds
+// or files may not be removed.
+struct WasiFile {
```

Comment:
Rather than using an RAII type for this single test, perhaps we should just insert the `fd_close` calls in the three places that are needed?

Reply:
What's wrong with RAII here?

Reply:
I'm also eager to remove RAII here. It makes the test logic somewhat clunky since we have to force nested scoping for the `drop` to fire. I cannot see why manually calling `fd_close` on all needed resources should be inferior to using the RAII type.

Reply:
Okay, I'll remove RAII.

Reply:
Yeah, I wouldn't say there's anything wrong with it by itself, just that it's a little bit of implementation that only saves three `fd_close` calls in a single test case.